### PR TITLE
Core update

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@Particular/azure-maintainers review please

File: src/DataBus/BlobStorageDataBus.cs [hunk 1]
```diff
@@ -30,29 +31,30 @@ public BlobStorageDataBus(CloudBlobContainer container)
...
 
-        public string Put(Stream stream, TimeSpan timeToBeReceived)
+        public Task<string> Put(Stream stream, TimeSpan timeToBeReceived)
         {
             var key = Guid.NewGuid().ToString();
             var blob = container.GetBlockBlobReference(Path.Combine(BasePath, key));
             SetValidUntil(blob, timeToBeReceived);
             UploadBlobInParallel(blob, stream);
-            return key;
+            return Task.FromResult(key);
```
Comment (SeanFeldman) [inline src/DataBus/BlobStorageDataBus.cs]: Since we have to return a task anyways, should we convert `UploadBlobInParallerl` to async by returning a task created by `CloudBlockBlob.UploadFromStreamAsync()`?

File: src/DataBus/BlobStorageDataBus.cs [hunk 2]
```diff
@@ -30,29 +31,30 @@ public BlobStorageDataBus(CloudBlobContainer container)
...
 
-        public Stream Get(string key)
+        public Task<Stream> Get(string key)
         {
-            var stream = new MemoryStream();
+            Stream stream = new MemoryStream();
             var blob = container.GetBlockBlobReference(Path.Combine(BasePath, key));
             DownloadBlobInParallel(blob, stream);
-            return stream;
+            return Task.FromResult(stream);
```
Comment (SeanFeldman) [inline src/DataBus/BlobStorageDataBus.cs]: Since we have to return a task anyways, should we convert `DownloadBlobInParallel` to async by returning a task created by `CloudBlockBlob.DownloadToStreamAsync()`?

File: src/DataBus/AzureDataBusDefaults.cs [hunk 3]
```diff
@@ -11,6 +10,6 @@ public class AzureDataBusDefaults
         public const string DefaultConnectionString = "UseDevelopmentStorage=true";
```
Comment (SeanFeldman) [inline src/DataBus/AzureDataBusDefaults.cs]: We should probably ditch it since we want to stop people relying on the local emulator by default and explicitly provide connection string. Where you planning to make this a separate PR @SimonCropp? If yes, please disregard my comment.

Comment (SimonCropp) [inline src/DataBus/AzureDataBusDefaults.cs]: there are a number of things i have seen which i think should be cleaned up. but i would rather do them on diff PRs and have conversations about each

File: src/DataBus/BlobStorageDataBus.cs (hunk 2 above)
Comment (SimonCropp) [inline src/DataBus/BlobStorageDataBus.cs]: done

File: src/DataBus/BlobStorageDataBus.cs (hunk 1 above)
Comment (SimonCropp) [inline src/DataBus/BlobStorageDataBus.cs]: done

**Comment by SimonCropp** [comment]
comments addressed. ping me if u see anything else

File: src/DataBus/AzureDataBusDefaults.cs (hunk 3 above)
Comment (SeanFeldman) [inline src/DataBus/AzureDataBusDefaults.cs]: +1

**Comment by SeanFeldman** [comment]
Thank you @SimonCropp
